src/catalog/catalog.cpp

```diff
@@ -24,19 +25,29 @@ Catalog::Catalog(transaction::TransactionManager *txn_manager, transaction::Tran
   CATALOG_LOG_TRACE("=======Finished Bootstrapping ======");
 }
 
-void Catalog::CreateDatabase(transaction::TransactionContext *txn, const std::string &name) {
+CatalogAccessor Catalog::GetAccessor(transaction::TransactionContext *txn, std::string database) {
+  auto db_handle = GetDatabaseHandle();
+  auto db_entry = db_handle.GetDatabaseEntry(txn, database);
+  return GetAccessor(txn, db_entry->GetOid());
+}
+
+db_oid_t Catalog::CreateDatabase(transaction::TransactionContext *txn, const std::string &name) {
   db_oid_t new_db_oid = db_oid_t(GetNextOid());
   Catalog::AddEntryToPGDatabase(txn, new_db_oid, name);
   BootstrapDatabase(txn, new_db_oid);
+  return new_db_oid;
 }
 
-void Catalog::DeleteDatabase(transaction::TransactionContext *txn, const std::string &db_name) {
+bool Catalog::DeleteDatabase(transaction::TransactionContext *txn, const std::string &db_name) {
   // get database handle
   auto db_handle = GetDatabaseHandle();
   auto db_entry = db_handle.GetDatabaseEntry(txn, db_name);
+  if (db_entry == nullptr) return false;
+
   auto oid = db_entry->GetOid();
   // remove entry from pg_database
-  db_handle.DeleteEntry(txn, db_entry);
+  bool ret = db_handle.DeleteEntry(txn, db_entry);
+  if (!ret) return false;
```

Comment:
I feel like this the lack of an error message may cause confusion when this returns `false`. Can you at least add some debug messages?